api/Controllers/FacilitiesController.cs

```diff
@@ -0,0 +1,169 @@
+            {
+                return await GetCheckedHotelFacilitiesAsync(facilityParams.HotelId ?? -1);
+            }
+            return await _hotelsDb.Facilities
+                .ProjectTo<Facility>(_mapper.ConfigurationProvider)
+                .ToArrayAsync();
+        }
+
+        [HttpPost]
+        [Authorize(Roles = Role.Admin)]
+        public async Task<IActionResult> AddFacility([FromBody] FacilityToAdd request)
+        {
+            FacilityEntity facility = _mapper.Map<FacilityEntity>(request);
+            if (!await CheckIfFacilityUniqueAsync(facility))
+            {
+                return BadRequest($"Facility with {facility.Name} name for {facility.Realm} already exists");
+            }
+            await _hotelsDb.AddAsync(facility);
+            await _hotelsDb.SaveChangesAsync();
+            return CreatedAtAction(nameof(GetFacility), new { facilityId = facility.Id }, facility.Id);
+        }
+
+        [Route("{facilityId}")]
+        [HttpDelete]
+        [Authorize(Roles = Role.Admin)]
+        public async Task<IActionResult> DeleteFacility([FromRoute] int facilityId)
+        {
+            FacilityEntity facility = await GetFacilityEntityAsync(facilityId);
+            _hotelsDb.Facilities.Remove(facility);
+            await _hotelsDb.SaveChangesAsync();
+            return NoContent();
+        }
+
+        [Route("{facilityId}")]
+        [HttpPut]
+        [Authorize(Roles = Role.Admin)]
+        public async Task<IActionResult> ChangeFacility([FromRoute] int facilityId, [FromBody] FacilityToEdit request)
+        {
+            FacilityEntity facilityEntity = await GetFacilityEntityAsync(facilityId);
+            if (facilityEntity == null)
+            {
+                return BadRequest("Such facility does not exist");
+            }
+            _mapper.Map(request, facilityEntity);
+            await _hotelsDb.SaveChangesAsync();
+            return NoContent();
+        }
+
+        private async Task<Facility[]> GetCheckedHotelFacilitiesAsync(int hotelId)
+        {
+            if (!await CheckIfHotelExistsAsync(hotelId))
+            {
+                return Array.Empty<Facility>();
+            }
+            Facility[] facilities = await _hotelsDb.Facilities
+                .Where(facility => facility.Realm == Realm.Hotel)
+                .ProjectTo<Facility>(_mapper.ConfigurationProvider)
+                .ToArrayAsync();
+            facilities = await MarkAsCheckedForHotelAsync(facilities, hotelId);
+            return facilities;
+        }
+
+        private async Task<Facility[]> MarkAsCheckedForHotelAsync(Facility[] facilities, int hotelId)
+        {
+            foreach (Facility facility in facilities)
+            {
+                if (await _hotelsDb.FacilityHotel.AnyAsync(fh => fh.FacilityId == facility.Id && fh.HotelId == hotelId))
+                {
+                    facility.Checked = true;
+                }
+            }
```

Comment:
Is there any problem here?

Reply:
I have changed this realization. hope u like it